docs/guides/dropout.rst

```diff
@@ -0,0 +1,147 @@
+Applying dropout
+================
+
+This guide provides an overview of how to apply
+`dropout <https://jmlr.org/papers/volume15/srivastava14a/srivastava14a.pdf>`__
+using :meth:`flax.linen.Dropout`
+
+Dropout is a stochastic regularization technique that randomly removes hidden
+and visible units in a network.
+
+Throughout the guide, you will be able to compare code examples with and without
+Flax ``Dropout``.
+
+.. testsetup::
+
+  import flax.linen as nn
+  import jax.numpy as jnp
+  import jax
+  import optax
+
+Split the PRNG
+**************
+
+Dropout is a random operation, requiring a PRNG state. Flax (like JAX) uses
+`Threefry <https://github.com/google/jax/blob/main/docs/jep/263-prng.md>`__ PRNG
+that is splittable.
+
+**Note:** Recall that JAX has an explicit way of giving you PRNG keys:
+you can fork the main PRNG state (such as ``key = jax.random.PRNGKey(seed=0)``)
```

Comment:
I agree that it's not common to have the argument name, but just in case some users wonder what `0` stands for (it could be any number), we should keep this IMO.